src/Service.Host/client/src/components/SerialNumbers.js

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState, useRef, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { Dropdown, Loading, Title, SnackbarMessage } from '@linn-it/linn-form-components-library';
+import {
+    TextField,
+    InputAdornment,
+    Typography,
+    Table,
+    TableHead,
+    TableRow,
+    TableCell,
+    TableBody
+} from '@material-ui/core';
+import { getSernosNote } from '../selectors/sernosNotesSelectors';
+import SerialNumber from './SerialNumber';
+import Page from '../containers/Page';
+
+function SerialNumbers({
+    items,
+    loading,
+    fetchItems,
+    sernosNotes,
+    snackbarVisible,
+    sernosNoteLoading,
+    sernosNotesLoading,
+    addSernosNote,
+    updateSernosNote,
+    setSnackbarVisible
+}) {
+    const [searchTerm, setSearchTerm] = useState('');
+    const [salesArticles, setSalesArticles] = useState([]);
+    const [selectedArticle, setSelectedArticle] = useState('');
+    const [debounceTimer, setDebounceTimer] = useState(null);
+
+    const savedFetchItems = useRef();
+    const savedDebounceTimer = useRef();
+
+    useEffect(() => {
+        savedFetchItems.current = fetchItems;
+    }, [fetchItems]);
+
+    useEffect(() => {
+        savedDebounceTimer.current = debounceTimer;
+    }, [debounceTimer]);
+
+    useEffect(() => {
+        if (items.length) {
+            const articles = [];
+            items.map(
+                item => !articles.includes(item.articleNumber) && articles.push(item.articleNumber)
+            );
+            setSalesArticles(articles);
+            setSelectedArticle(articles[0] || '');
+        }
+    }, [items]);
+
+    useEffect(() => {
+        if (searchTerm) {
+            if (savedDebounceTimer.current) {
+                clearTimeout(savedDebounceTimer.current);
+            }
+
+            setDebounceTimer(
+                setTimeout(() => savedFetchItems.current('sernosNumber', searchTerm), 500)
+            );
+        } else if (savedDebounceTimer.current) {
+            clearTimeout(savedDebounceTimer.current);
+        }
+    }, [searchTerm]);
+
```

Comment:
Typeahead implemented with useEffect. I'd like to move this to a shared component